engine/component.ftl

```diff
@@ -126,18 +126,59 @@
 [/#macro]
 
 [#-- Not for general use - framework only --]
-[#assign coreComponentChildConfiguration = [
-    {
-        "Names" : ["Export"],
-        "Types" : ARRAY_OF_STRING_TYPE,
-        "Default" : []
-    },
-    {
-        "Names" : ["DeploymentUnits"],
-        "Types" : ARRAY_OF_STRING_TYPE,
-        "Default" : []
-    }
-] ]
+[#assign coreComponentDeploymentUnitConfiguration = 
+    [
+        {
+            "Names" : ["DeploymentUnits", "deployment:Unit"],
+            "Description" : "An singleton instance of the component, Instance & Version configuration.",
+            "Types" : ARRAY_OF_STRING_TYPE,
+            "Default" : []
+        }
+    ]
+]
+
+[#assign coreComponentChildConfiguration = 
+    [
+        {
+            "Names" : ["Export"],
+            "Types" : ARRAY_OF_STRING_TYPE,
+            "Default" : []
+        },
+        {
+            "Names" : "deployment:Priority",
+            "Description" : "The Deployment Priority",
+            "Types" : NUMBER_TYPE,
+            "Default" : 5
+        },
+        {
+            "Names" : [ "deployment:Group", "DeploymentGroup" ],
+            "Description" : "The component Deployment Group",
+            "Types" : STRING_TYPE
+        },
```

Comment:
I don't think these should be included as part of this. They are now only added where they are actually used, rather than being permitted everywhere. Where were you seeing these as not being set?